From 23017dc94feda78e93dd48aac955efb49dc846e9 Mon Sep 17 00:00:00 2001
From: Yu Li <liyu@apache.org>
Date: Sat, 9 Jul 2016 01:05:10 +0800
Subject: [PATCH 463/471] HBASE-16201 fix a NPE issue in RpcServer

(cherry picked from commit b99efe65e900299271b1e2a0c5feabd23930eb70)

Change-Id: I79707378c9a894d7993ad4c86a23ff75d79de3d6
Reason: Bug
Ref: CDH-67494
Author: Yu Li
---
 .../org/apache/hadoop/hbase/ipc/RpcServer.java     |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
index 66c67fa..e9a6035 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
@@ -2225,7 +2225,13 @@ public class RpcServer implements RpcServerInterface, ConfigurationObserver {
       // The above callBlockingMethod will always return a SE.  Strip the SE wrapper before
       // putting it on the wire.  Its needed to adhere to the pb Service Interface but we don't
       // need to pass it over the wire.
-      if (e instanceof ServiceException) e = e.getCause();
+      if (e instanceof ServiceException) {
+        if (e.getCause() == null) {
+          LOG.debug("Caught a ServiceException with null cause", e);
+        } else {
+          e = e.getCause();
+        }
+      }
 
       // increment the number of requests that were exceptions.
       metrics.exception(e);
-- 
1.7.9.5

